spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfiguration.java

```diff
@@ -49,6 +50,7 @@
  */
 @Configuration
 @ConditionalOnClass({ DataSource.class, EmbeddedDatabaseType.class })
+@ConditionalOnSingleCandidate(DataSource.class)
```

Comment:
Not sure this makes sense as a few lines later there is `@ConditionalOnMissingBean({ DataSource.class, XADataSource.class })` which will now never happen.